pkg/scalers/scaler.go

```diff
@@ -209,3 +211,72 @@ func GenerateMetricInMili(metricName string, value float64) external_metrics.Ext
 		Timestamp:  metav1.Now(),
 	}
 }
+
+// getParameterFromConfigV2 returns the value of the parameter from the config
+func getParameterFromConfigV2(config *ScalerConfig, parameter string, useMetadata bool, useAuthentication bool, useResolvedEnv bool, isOptional bool, defaultVal string, targetType reflect.Type) (interface{}, error) {
+	foundCount := 0
+	var foundVal string
+	var convertedVal interface{}
+	var foundErr error
+	
+	if val, ok := config.AuthParams[parameter]; useAuthentication && ok && val != "" {
+		foundCount ++
+		foundVal = val
+	}
+	if val, ok := config.TriggerMetadata[parameter]; ok && useMetadata && val != "" {
+		foundCount ++
+		foundVal = val
+	}
+	if val, ok := config.TriggerMetadata[fmt.Sprintf("%sFromEnv", parameter)]; ok && useResolvedEnv && val != "" {
+		foundCount ++
+		foundVal = val
+	}
```

Comment:
WDYT if we count the occurrence, although we don't use that source?

```suggestion
	if val, ok := config.AuthParams[parameter]; ok && val != "" {
		foundCount++
		if useAuthentication {
			foundVal = val
		}
	}
	if val, ok := config.TriggerMetadata[parameter]; ok && val != "" {
		foundCount++
		if useMetadata {
			foundVal = val
		}
	}
	if envFromVal, envFromOk := config.TriggerMetadata[fmt.Sprintf("%sFromEnv", parameter)]; envFromOk {
		if val, ok := config.ResolvedEnv[envFromVal]; ok && val != "" {
			foundCount++
			if useResolvedEnv {
				foundVal = val
			}
		}
	}
```

Reply:
@JorTurFer yes we can definitely do that. Just one small part on this section 
```golang
if envFromVal, envFromOk := config.TriggerMetadata[fmt.Sprintf("%sFromEnv", parameter)]; envFromOk {
		if val, ok := config.ResolvedEnv[envFromVal]; ok && val != "" {
			foundCount++
			if useResolvedEnv {
				foundVal = val
			}
		}
```
This is the way to go right `config.ResolvedEnv[envFromVal]` I always thought we just return directly the value of `envFromVal` 🤔

Reply:
No, `envFromVal` is the env name that KEDA has too look for in workload pod and that's why with using envFromVal we go through `config.ResolvedEnv` (because `config.ResolvedEnv` is a map with all the env vars and their values)